spec/src/main/asciidoc/app-programming-model.adoc

```diff
@@ -660,3 +661,57 @@ metricRegistry.register(m, me, new Tag("colour","blue"));
 ----
 
 A default implementation `DefaultMetadata` is provided in the API for convenience.
+
+=== Using the NamedMetadata annotation
+Because all metrics under the same name are required to have the same metadata, it could get cumbersome for developers
+who want to declare many metrics with the same name via annotations - if the metrics declare any other metadata
+apart from the name, developers have to repeat the exact same metadata in all the annotations. To prevent the need
+for such duplication, the `@NamedMetadata` annotation was introduced.
+
+The `@NamedMetadata` annotations serves as a single point of declaration for a global metadata definition that can be shared
+by multiple metrics. Metrics that reference the same metadata definition can be registered under same name,
+but it's possible to use this for metrics with different names as well). This annotation can be places on any
```

Comment:
since `@NamedMetadata` contains the `metricName`, how can it be used for metrics with different names?  Is `metricName` required?  If so, seems like it could only be used for one metric name... if not, more complexity for dealing with case where `metricName` is missing.

Reply:
I see below that metricName is required... so how could this be used for metrics with different names?

Reply:
Hm, good point, I didn't realize that. Will remove this sentence